functional-tests/run_test.sh

```diff
@@ -11,9 +11,8 @@ fi
 
 # Conditionally run cargo build based on PROVER_TEST
 if [ ! -z $PROVER_TEST ]; then
-    echo "Running on prover mode"
-    # cargo build -F "prover"
-    cargo build -F "prover" --release
+    echo "Running on risc0-mock mode"
+    cargo build --release -F risc0-mock
```

Comment:
With this PR, we can run the current prover client tests in native mode right ?

Reply:
Yes, we can do that. Can we scope that to a different PR as this is already getting large.
When writing new functional tests, we also want to cover edge cases related with bridge deposits and withdraws as well.